src/tls/cockpit-certificate-ensure.c

```diff
@@ -162,6 +192,45 @@ certificate_and_key_clear (CertificateKeyPair *self)
   self->filename_for_errors = NULL;
 }
 
+static void
+certificate_and_key_write (const CertificateKeyPair *self,
+                           const char               *directory)
+{
+  int dirfd = open (directory, O_PATH | O_DIRECTORY | O_NOFOLLOW);
+  if (dirfd == -1)
+    err (EXIT_FAILURE, "open: %s", directory);
+
+  struct stat buf;
+  if (fstat (dirfd, &buf) != 0)
```

Comment:
I mean like, `open()` is documented as "-1 on fail".

`fstat()` is documented the same, but it's also documented "0 on success".

I guess the difference here is that `dirfd` might be (and almost certainly is) something other than 0, so of course we can't just check it for `== 0`.

Case by case, I guess I could argue that `fstat()` had better return exactly 0, and if it doesn't, I'd like to know about it.  As for `open()`, I guess I could check for `< 0`, but like, the manpage is pretty clear that `-1` is the error case, so that's just what I check for...

Total bikeshed/style thing, and I get your point... I just don't see myself coming around to it :)

Reply:
Heh, sure. But you know by now, we can't have a PR without any such bikeshedding! :stuck_out_tongue_winking_eye: